### PR TITLE
Fix `destination-s3` build

### LINKED ISSUE #26521: Destination S3: Fix docker build
https://github.com/airbytehq/airbyte/pull/26489 adds the ability for airbyte-ci to run additional scripts in the docker build. This is important for destination-s3, which needs to install the lzo compression library. Its script currently does not work (see https://github.com/airbytehq/airbyte/pull/26489/files#r1204221382). We should fix it so that destination-s3 can be published.

[Timebox this research to 1 day]

### PR DESCRIPTION
**Pull request by evantahler**
Closes https://github.com/airbytehq/airbyte/issues/26521

Reference for failing build https://github.com/airbytehq/airbyte/actions/runs/5387253975

Locally, `airbyte-ci connectors --name=destination-s3 build` does work fine...

```
╭─────────────────────────────────────────────────────────────────────────────────────────────────────── DESTINATION-S3 - BUILD RESULTS ───────────────────────────────────────────────────────────────────────────────────────────────────────╮
│                                        Steps results                                                                                                                                                                                         │
│ ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━┓                                                                                                                                                 │
│ ┃ Step                                                       ┃ Result     ┃ Finished after ┃                                                                                                                                                 │
│ ┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━┩                                                                                                                                                 │
│ │ Build destination-s3 docker image for platform linux/amd64 │ Successful │ 79s            │                                                                                                                                                 │
│ │ Build destination-s3 docker image for platform linux/arm64 │ Successful │ 23s            │                                                                                                                                                 │
│ │ Load airbyte/destination-s3:dev to the local docker host.  │ Successful │ 0s             │                                                                                                                                                 │
│ └────────────────────────────────────────────────────────────┴────────────┴────────────────┘                                                                                                                                                 │
╰───────────────────────────────────────────────────────────────────────────────────────── ⏲️  Total pipeline duration for destination-s3: 412 seconds ─────────────────────────────────────────────────────────────────────────────────────────╯
[14:52:03] INFO     Build: {"connector_technical_name": "destination-s3", "connector_version": "0.5.0", "run_timestamp": "2023-06-27T21:52:03.101838", "run_duration": 412.072144, "success": true, "failed_steps": [],          contexts.py:401
                    "successful_steps": ["BuildConnectorImage", "BuildConnectorImage", "LoadContainerToLocalDockerHost"], "skipped_steps": [], "gha_workflow_run_url": null, "pipeline_start_timestamp": 1687927511,
                    "pipeline_end_timestamp": 1687927923, "pipeline_duration": 412, "git_branch": "evan/destination-s3-build", "git_revision": "fd34fe3e2b3c037bd686e9a270dc91e881aeb826", "ci_context": "manual"}
```

### CODE REVIEW

**Review by edgao — APPROVED** [review]
lgtm pending pr checks I guess? [augustin's comment](https://github.com/airbytehq/airbyte/pull/26489/files#r1204221382) said the `sdk install java 8.0.342-librca` failed, so if that's not happening then we should be good to go :shrug:

**Comment by octavia-squidington-iii** [comment]
##  destination-s3 test report (commit [`8e0f8874ec`](https://github.com/airbytehq/airbyte/pull/27786/commits/8e0f8874ecfd1fb4c31e6b8f05a9db8f689ab8d2)) - ✅

⏲️  Total pipeline duration: 908 seconds

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-s3/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| Connector version increment check                                     | ✅        |
| QA checks                                                             | ✅        |
| Build connector tar                                                   | ✅        |
| Build destination-s3 docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                            | ✅        |
| Integration tests                                                     | ✅        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/evan_destination-s3-build/1687904216/8e0f8874ecfd1fb4c31e6b8f05a9db8f689ab8d2/destination-s3/0.5.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-s3 test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-s3 test report (commit [`ef6b83828e`](https://github.com/airbytehq/airbyte/pull/27786/commits/ef6b83828eb37d20c338c8da3ec4f3dcdc3c43f5)) - ✅

⏲️  Total pipeline duration: 826 seconds

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-s3/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| Connector version increment check                                     | ✅        |
| QA checks                                                             | ✅        |
| Build connector tar                                                   | ✅        |
| Build destination-s3 docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                            | ✅        |
| Integration tests                                                     | ✅        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/evan_destination-s3-build/1687906965/ef6b83828eb37d20c338c8da3ec4f3dcdc3c43f5/destination-s3/0.5.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-s3 test
```

File: airbyte-integrations/bases/base-normalization/build.gradle
Comment (evantahler) [inline airbyte-integrations/bases/base-normalization/build.gradle:62]: unrelated to this PR, but I was annoyed.  I added this log statement in https://github.com/airbytehq/airbyte/pull/26677, but I didn't realize it would log on *every* gradle run, even those that have nothing to do with normalization 😓

**Comment by octavia-squidington-iii** [comment]
##  destination-s3 test report (commit [`5ea95c0ae0`](https://github.com/airbytehq/airbyte/pull/27786/commits/5ea95c0ae0f1c192714a2241bcc2ded65d3bafda)) - ✅

⏲️  Total pipeline duration: 866 seconds

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-s3/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| Connector version increment check                                     | ✅        |
| QA checks                                                             | ✅        |
| Build connector tar                                                   | ✅        |
| Build destination-s3 docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                            | ✅        |
| Integration tests                                                     | ✅        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/evan_destination-s3-build/1687910028/5ea95c0ae0f1c192714a2241bcc2ded65d3bafda/destination-s3/0.5.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-s3 test
```

**Comment by evantahler** [comment]
cc @alafanechere 
I'm going to merge this PR, as we are getting green test reports for the connector (yay!).  However, there's something odd with the test reporting:
* The green test reports link to the latest commmits
* `Connectors tests / Connectors CI (pull_request)` seem to be waiting on a runner for the test
* The event timing is strange in the [workflow report](https://github.com/airbytehq/airbyte/actions/runs/5395240921). It says it ran for 1h 34min, but the action within only took 15min.  

Screenshots from right before my merge:

Screenshot 2023-06-27 at 5 11 30 PM

Screenshot 2023-06-27 at 5 11 33 PM

**Comment by octavia-squidington-iii** [comment]
##  destination-s3 test report (commit [`dab48992ef`](https://github.com/airbytehq/airbyte/pull/27786/commits/dab48992ef06ce0302160722f15e659191ff04c5)) - ❌

⏲️  Total pipeline duration: 912 seconds

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-s3/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| Connector version increment check                                     | ❌        |
| QA checks                                                             | ✅        |
| Build connector tar                                                   | ✅        |
| Build destination-s3 docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                            | ✅        |
| Integration tests                                                     | ✅        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/evan_destination-s3-build/1687912221/dab48992ef06ce0302160722f15e659191ff04c5/destination-s3/0.5.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-s3 test
```

**Pull request merged**
